controller/src/beerocks/master/tasks/association_handling_task.cpp

```diff
@@ -372,6 +372,18 @@ void association_handling_task::finalize_new_connection()
      */
     if (!database.get_node_handoff_flag(sta_mac)) {
         if (database.get_node_type(sta_mac) == beerocks::TYPE_CLIENT) {
+            auto client_mac = tlvf::mac_from_string(sta_mac);
+            // If stay_on_initial_radio is enabled and initial_radio is not set yet, set to parent radio mac (not bssid)
+            if ((database.get_client_stay_on_initial_radio(client_mac) == eTriStateBool::ENABLE) &&
+                (database.get_client_initial_radio(client_mac) == network_utils::ZERO_MAC)) {
+                auto bssid            = database.get_node_parent(sta_mac);
+                auto parent_radio_mac = database.get_node_parent_radio(bssid);
+                if (!database.set_client_initial_radio(client_mac,
+                                                       tlvf::mac_from_string(parent_radio_mac))) {
+                    LOG(WARNING) << "Failed to set client " << client_mac << "  initial radio to "
+                                 << parent_radio_mac;
+                }
+            }
```

Comment:
I think it will be a good idea to add a comment about what you are doing here, and why.

Reply:
add a comment describing the reason we set the initial-radio here and what is the pre-condition to setting it.